arrows/vxl/average_frames.cxx

```diff
@@ -479,93 +485,69 @@ class average_frames::priv
   bool output_variance{ false };
   double variance_scale{ 0.0 };
 
-  // The actual frame averager
-  frame_averager_byte_sptr byte_averager;
-  frame_averager_float_sptr float_averager;
+  // The actual frame averagers
+  using averager_ptr = std::unique_ptr< online_frame_averager_base >;
+  std::unordered_map< vil_pixel_format, averager_ptr > frame_averager;
 
   // --------------------------------------------------------------------------
   // Load model, special optimizations are in place for the byte case
-  void
-  load_model( bool is_byte = true )
+  template < typename PixType >
+  online_frame_averager< PixType >*
+  load_model()
   {
-    if( ( is_byte && byte_averager ) || ( !is_byte && float_averager ) )
-    {
-      return;
-    }
-
-    switch( type )
+    auto& averager = frame_averager[ vil_pixel_format_of( PixType{} ) ];
+    if( !averager )
     {
-      case AVERAGER_window:
+      switch( type )
       {
-        if( is_byte )
+        case AVERAGER_window:
         {
-          byte_averager.reset(
-            new windowed_frame_averager< vxl_byte >{ round, window_size } );
+          averager.reset( new windowed_frame_averager< PixType >{} );
+          break;
         }
-        else
+        case AVERAGER_cumulative:
         {
-          float_averager.reset(
-            new windowed_frame_averager< double >{ round, window_size } );
+          averager.reset( new cumulative_frame_averager< PixType >{ round } );
+          break;
         }
-        break;
-      }
-      case AVERAGER_cumulative:
-      {
-        if( is_byte )
+        case AVERAGER_exponential:
         {
-          byte_averager.reset(
-            new cumulative_frame_averager< vxl_byte >{ round } );
+          if( exp_weight <= 0 || exp_weight >= 1 )
```

Comment:
This feels like a duplicate check with what is done in `check_config`. Is this since someone could use this class outside of the arrow, or are there situations where processing code can be run without passing `check_config`?

Reply:
Assuming you mean `check_configuration`... AFAICT that method is a convenience that is hardly ever used in practice (outside of unit tests).